### ISSUE TITLE
SelectableText.rich() with VoiceOver does not allow user to focus on tappable links.

### ISSUE THREAD

**Issue by parruck** [flutter/flutter#77219]
Same issue as https://github.com/flutter/flutter/issues/20643, except for the SelectableText.rich() widget. 

The TaggedText widget returns a RichText or SelectableText.rich() widget, based on if selectableText is set to true. In the case that it returns a RichText, the accessibility works as intended, but in the other case, the bug is exactly the same as the linked one: users cannot focus on the links in the text.

For instance: 
          TaggedText(
            selectableText: true,
            content: 'Example of string with <a href="#"> link </a>'
            onTapLink: someFunction (),
),

In this case navigating the app by voiceover does not allow the users to focus on the tappable link.

Internal: b/172313464

**Comment by darshankawar** [comment]
@parruck Hi.

Tried a sample code based on description above and with Talkback ON, the text which has a link as a sub-part, reads the entire text when focused, as below:

<img width="546" alt="Screenshot 2021-03-04 at 4 13 34 PM" src="https://user-images.githubusercontent.com/67046386/109952475-e0cffb80-7d04-11eb-94fd-6f5532a8fdd2.png">

Where `test` underlined is a hyperlink.

```
home: Scaffold(
          body: Center(
              child: SelectableText.rich(
        TextSpan(
          children: <TextSpan>[
            TextSpan(text: 'This', style: TextStyle(color: Colors.blue)),
            TextSpan(text: 'is', style: TextStyle(color: Colors.black)),
            TextSpan(
                text: 'test',
                style: TextStyle(
                    color: Colors.blue, decoration: TextDecoration.underline),
                recognizer: TapGestureRecognizer()
                  ..onTap = () async {
                    var url = "https://flutter.dev/";
                    if (await canLaunch(url)) {
                      await launch(url);
                    } else {
                      throw 'Could not launch $url';
                    }
                  }),
          ],
        ),
        style: TextStyle(
            color: Colors.blue, fontWeight: FontWeight.bold, fontSize: 48),
        textAlign: TextAlign.center,
      ))),
```

Is this a correct interpretation of the issue ?

**Comment by parruck** [comment]
Yes, Darshankawar - this is accurate, thank you! The text should ideally be read as 'This is, text' and the next element should be 'test, link'.

**Comment by darshankawar** [comment]
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.0.0, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.0.0 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 60bd88df91 (14 hours ago), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API
      27)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.4
      19E2269 darwin-x64
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      88.0.4324.192

! Doctor found issues in 1 category.

```

</details>
